service/history/task/standby_task_util.go

```diff
@@ -158,7 +158,11 @@ func getHistoryResendInfo(
 	mutableState execution.MutableState,
 ) (*historyResendInfo, error) {
 
-	currentBranch, err := mutableState.GetVersionHistories().GetCurrentVersionHistory()
+	versionHistories := mutableState.GetVersionHistories()
+	if versionHistories == nil {
+		return nil, &shared.BadRequestError{Message: "versionHistories is empty, which is required for NDC feature. It's probably from deprecated 2dc workflows"}
```

Comment:
I guess this will make the task stuck forever? Probably also update the task error handling logic to not retry on this error?

Reply:
updated in the task.go